src/Modules/ModulesGraded.jl

```diff
@@ -2567,3 +2567,89 @@ function ideal_as_module(I::MPolyIdeal)
   e1 = F[1]
   return sub(F, [x * e1 for x = gens(I)], :module)
 end
+
+
+
+##########################################################################
+##### Twists
+##########################################################################
+
+@doc raw"""
+    twist(M::ModuleFP{T}, g::GrpAbFinGenElem) where {T<:MPolyDecRingElem}
+
+Return the twisted module `M(g)`.
+
+    twist(M::ModuleFP{T}, W::Vector{<:IntegerUnion}) where {T<:MPolyDecRingElem}
+
+Given a module `M` over a $\mathbb Z^m$-graded polynomial ring and a vector `W` of $m$ integers, 
+convert `W` into an element `g` of the grading group of the ring and proceed as above.
+
+    twist(M::ModuleFP{T}, d::IntegerUnion) where {T<:MPolyDecRingElem}
+
+Given a module `M` over a $\mathbb Z$-graded polynomial ring and an integer `d`, 
+convert `d` into an element `g` of the grading group of the ring and proceed as above.
+
+# Examples
+```jldoctest
+julia> R, (x, y) = graded_polynomial_ring(QQ, ["x", "y"]);
+
+julia> I = ideal(R, [zero(R)])
+ideal(0)
+
+julia> M = quotient_ring_as_module(I)
+Graded submodule of R^1
+1 -> e[1]
+represented as subquotient with no relations
+
+julia> degree(gen(M, 1))
+[0]
+
+julia> N = twist(M, 2)
+Graded submodule of R^1
+1 -> e[1]
+represented as subquotient with no relations
+
+julia> degree(gen(N, 1))
+[-2]
+
+```
+"""
+function twist(M::ModuleFP{T}, g::GrpAbFinGenElem) where {T<:MPolyDecRingElem}
+ error("Not implemented for the given type")
```

Comment:
I am not going to block this PR over this, but, normally we strive to indent code with (at least) two spaces.

(But of course we are already widely inconsistent about this, which is why I am not asking for it to be changed here and now, just wanted to point it out)